### ISSUE TITLE
Research Request - Add Amtrak routes in for rail HQTA

### ISSUE THREAD

**Issue by tiffanychu90** [cal-itp/data-analyses#1035]
Complete the below when receiving a research request, and continue to add to this issue as you receive additional details and produce deliverables. Be sure to also add the appropriate project-level label to this issue (eg gtfs-rt, DLA).

## Research Question
**Single sentence** description:  Right now, Amtrak is excluded from HQTA analysis. Add some routes that run in CA (Capitol Corridor, San Joaquins, Pacific Surfliner).
  
**Detailed** description: We can do this in a similar way to BRT...where certain routes are selected from all bus routes. [Slack thread](https://cal-itp.slack.com/archives/C02H6JUSS9L/p1708044377243689). While we're at it, these scripts are due for a light refactor...some of the logic is getting hard to follow.

## How will this research be used?
Open data portal

**Comment by tiffanychu90** [comment]
See [Slack thread](https://cal-itp.slack.com/archives/C02KH3DGZL7/p1708710990047189). 

TLDR:  `Amtrak Schedule` may belong to `MTC 511 for Amtrak San Joaquins`...but this doesn't seem to show up as a sub-feed. This [filter in gtfs_utils_v2](https://github.com/cal-itp/data-analyses/blob/main/_shared_utils/shared_utils/gtfs_utils_v2.py#L120) is in place, and that is mostly desired behavior (unless Amtrak should be the only exception).

Decision points: 
* should precursor feeds be allowed? Only Amtrak?
* is it ok that we do not find `Bay Area 511 Amtrak San Joaquins`?

```
SELECT DISTINCT
  *
FROM cal-itp-data-infra.mart_transit_database.dim_gtfs_datasets
WHERE name = 'Amtrak Schedule' 
ORDER BY _valid_to DESC
```
shows that `regional_feed_type = "Regional Precursor Feed"`, which is filtered out in `gtfs_utils_v2`, where we don't allow
